Libraries/Components/View/ViewPropTypes.js

```diff
@@ -319,6 +319,20 @@ type AndroidViewProps = $ReadOnly<{|
    * @platform android
    */
   nextFocusUp?: ?number,
+
+  /**
+   * Whether this `View` should be clickable with a non-touch click, eg. enter key on a hardware keyboard.
+   *
+   * @platform android
+   */
+  clickable?: boolean,
```

Comment:
I want to share some details about some of the work happening inside Facebook and how it relates to this change.

* Supporting multiple input types is a problem for all platforms.
* The web platform has designed APIs for this purpose.
* The concept of "click" on the web is essentially legacy, and touch events still trigger `onClick`.
* React Native for Web supports multiple input types but relies on many more events than `onClick` (i.e., `onMouseDown`, `onKeyPress`, etc.).
* Internally we're experimenting with a [new React event system](https://github.com/facebook/react/issues/15257). On the web that relies on Pointer Events and would eschew the need to use `onClick`. The general idea is to move away from input-specific event types and instead use `onPress`, etc., with the event object containing information about the pointer type that was used.
* Internally we're experimenting with web implementations of `View` and `Text` that shed all the event props found in React Native (in favour of the new event system) and introduce a `focusable` prop to mark a View as being able to receive (or not) user-initiated focus by any input type (all Views are programmatically focusable). In this way you could set `focusable={false}` and retain all the mouse/touch functionality but hide the View from inputs like keyboards that first need to focus an element before it can be interacted with. That's an important pattern on the web where you might want to make something like an avatar interactive to pointers, but hidden from keyboards if the functionality is redundant, i.e., if there's also an interactive user name that performs the same action.

Given all that I'd suggest we do something like rename `clickable` to `focusable` and decouple it from `onClick`. If `onClick` here is meant to be used for mouse as well as keyboard, that's probably a good enough prop for now to unblock other input types on Android. We'll hopefully address this properly in RN in the future with the new event system.

Reply:
Thanks for the write up!

About the decouple of `focusable` from `onClick`, what do you suggest?

The Android default onClick listener works for all input types 'keyboard', 'touch', 'mouse', which is what we need support for. Before this change, only 'touch' and 'mouse' events would work.

Reply:
Naming this "focusable" could definitely lead to some confusion here due to Android's already confusing separate concepts of keyboard focusability and accessibility focusability.

Setting "focusable=true" on a standard Android view will set it to be both keyboard and accessibility focusable, but if this is property is meant to only trigger keyboard focusability, we should probably be more explicit in its naming.

Reply:
> About the decouple of focusable from onClick, what do you suggest?

I was thinking that the `focusable` feature is something we need independent of how events are handled. So in the future we might not have event handlers on `View` but we'll still need to mark views as being able to receive user-initiated focus following a mouse, touch, or keyboard event.

> Setting "focusable=true" on a standard Android view will set it to be both keyboard and accessibility focusable, but if this is property is meant to only trigger keyboard focusability, we should probably be more explicit in its naming.

What is "accessibility focusable" on Android? The way we're using this on web is to allow the element to receive focus following any user action from any pointer device. It happens to also be required to support interactions on a view (with seems to overlap with this PR for Android).

Reply:
To answer what "accessibility focusable" is requires a bit of a history lesson, so bear with me. 

In the early days of Android many devices had trackballs or d-pads as an alternate non-touch way to interact with the device.  To the system, these keys worked much like external keyboards. To allow your UI to be interactive you had to make sure it was focusable in "non touch mode", and the best way to do this was to set focusable="true".  You would generally want all interactive elements to be focusable in this way, but wouldn't want non-interactive elements to be focusable, since it would be annoying to have to press keys to get past all of the non-interactive content.  

Then later on Google decided to add a real accessibility system to Android, and built the Talkback screen reader as well as other accessibility services like Switch Access and Select to Speak.  Some of these services (like Talkback) can only really be useful if you can focus on non-interactive content, like plain text.  Otherwise, this content could be totally hidden from a user with a vision impairment. This meant that they couldn't simply use the same focus system used for D-Pad's and keyboards, and they had to come up with a way that you could have an element be focusable for accessibility, but *not* focusable for keyboard users.  This is what I mean when I say "accessibility focusable".  

Unfortunately, they didn't simply add a property like accessibilityFocusable and instead derived whether or not something would be accessibility focusable by looking at the combination of multiple different properties (click handlers, the focusable property, if the view was a leaf node, if it had focusable ancestors, etc.).  It wasn't until Android Pie that they cleaned this up and added `setScreenReaderFocusable` to the View class.

Reply:
Thank you! It sounds to me like the `focusable` concept is equivalent to its meaning on Android, as it's intended for interactive elements. Setting `focusable` to `false` would mean the element is no longer accessible to keyboards. On the web it would still be accessible to screen readers, and hiding it from them would need `accessibilityState.hidden` to be `true` as well.

Reply:
Any more thoughts?

Reply:
Thanks for all the info and great discussion above.

I think it makes sense to rename this prop to something more correct. What about `keyClickable` or `externalClickable`?

Reply:
Neither of those names seem quite right. FWIW I noticed SwiftUI also uses `focusable` for essentially the same concept - https://developer.apple.com/documentation/swiftui/view/3278564-focusable - with the difference being that in React (at least on web, for now) if `focusable` is set to false on a View it is still able to receive programmatic focus for accessibility reasons. So `focusable` is about whether a view can receive user-initiated focus from a pointer or keyboard

Reply:
On our platforms, we also use focusable and that’s what @jayphelps (formerly of Netflix and ThisDot Labs) recommended in his draft RFC on this topic. Besides keyboard, and 5-way controls, this also maps well to gaze (AR/VR/XR) and 6DOF controls in my experience. It’s really awesome to see this ideas make it into React itself!

Reply:
Thanks, I think it makes sense to call it `focusable`. Who should do the work? Do we just update this PR?

Reply:
Please send a new PR that changes everything. We can do it all in a single commit :)

Reply:
If you'd like to write the patch that would be great! A new PR is best since this one has already been merged.

Reply:
Hey @sweggersen: just wanted to ping you and ask whether you could send a PR and rename this to `focusable`?

Reply:
It's on my list. I'll do it this week.

Reply:
It's on my list. I'll do it this week.